src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/MigrationService.kt

```diff
@@ -12,24 +13,30 @@ import java.time.Duration
 import java.time.LocalDateTime
 
 abstract class MigrationService<FILTER : Any, NOMIS_ID : Any, NOMIS_ENTITY : Any, MAPPING : Any>(
-  internal val queueService: MigrationQueueService,
-  internal val migrationHistoryService: MigrationHistoryService,
   internal val mappingService: MigrationMapping<MAPPING>,
-  internal val telemetryClient: TelemetryClient,
-  internal val auditService: AuditService,
   internal val migrationType: MigrationType,
   private val pageSize: Long,
   private val completeCheckDelaySeconds: Int,
   private val completeCheckCount: Int,
 ) {
 
+  @Autowired
+  protected lateinit var queueService: MigrationQueueService
+
+  @Autowired
+  protected lateinit var migrationHistoryService: MigrationHistoryService
+
+  @Autowired
+  protected lateinit var telemetryClient: TelemetryClient
+
+  @Autowired
+  protected lateinit var auditService: AuditService
```

Comment:
Autowire into superclass instead of autowiringinto every subclass and passing them in the constructor which adds lots of boilerplate to the constructor parameters.